### PR TITLE
Add assertions for pegout transaction created event

### PR DESCRIPTION
**Pull request by nathanieliov**
*rskj:master*

### CODE REVIEW

**Comment by nathanieliov** [comment]
pipeline:run

**Comment by nathanieliov** [comment]
pipeline:run

**Comment by nathanieliov** [comment]
pipeline:run

**Comment by nathanieliov** [comment]
pipeline:run

**Comment by nathanieliov** [comment]
pipeline:run

File: lib/tests/2wp.js [hunk 1]
```diff
@@ -175,6 +179,28 @@ const execute = (description, getRskHost) => {
             expect(batchPegoutCreatedEvent).to.not.be.null;
             expect(batchPegoutCreatedEvent.arguments.releaseRskTxHashes.includes(pegoutTransaction.transactionHash)).to.be.true;
           }
+
+          const isLovell700AlreadyActive = await Runners.common.forks.lovell700.isAlreadyActive();
+          if (isLovell700AlreadyActive) {
+            const pegoutTransactionCreatedEvent = await rskUtils.findEventInBlock(localRskTxHelper, 'pegout_transaction_created');
```
Comment (jeremy-then) [inline lib/tests/2wp.js:185]: Is it time to have these event names declared in an enum like structure? What do you think?

File: lib/tests/2wp.js [hunk 2]
```diff
@@ -175,6 +179,28 @@ const execute = (description, getRskHost) => {
...
+            expect(pegoutTransactionCreatedEvent).to.not.be.null;
+            const encodedUtxoOutpointValues = Buffer.from(removePrefix0x(pegoutTransactionCreatedEvent.arguments.utxoOutpointValues), 'hex');
+
+            const federationUtxoValues = activeFederationUtxosBeforePegout.reduce((map, utxo) => {
+              map[utxo.valueInSatoshis] = Buffer.from(new VarInt(utxo.valueInSatoshis).encode()).toString("hex");
+              return map;
+            }, {});
+
+            let offset = 0;
```
Comment (jeremy-then) [inline lib/tests/2wp.js:194]: Too much decoding logic in a test, don't you think? What if you take the `VarInt` decoding to an utility function that returns a list of decoded outpoint values and then you can compare it like:

```javascript
const outpointValues = getDecodedOutpointsValues(encodedUtxoOutpointValues);
//...
expect(outpointValues.every(value => federationUtxoValues.has(value)).to.be.true;
```

Or something like that.

File: lib/rsk-utils.js [hunk 3]
```diff
@@ -400,6 +398,19 @@ const findEventInBlock = async (rskTxHelper, eventName, fromBlockHashOrBlockNumb
...
+  const isLovell700AlreadyActive = await Runners.common.forks.lovell700.isAlreadyActive()
+
+  if (isLovell700AlreadyActive){
+    const bridgeTransactionParser = new BridgeTransactionParser(web3Client);
+    return bridgeTransactionParser.getBridgeTransactionsInThisBlock(blockHashOrBlockNumber);
+  }
+
+  const isFingerroot500AlreadyActive = await Runners.common.forks.fingerroot500.isAlreadyActive();
+  const bridgeTxParser = isFingerroot500AlreadyActive ? fingerrootBridgeTxParser : hopBridgeTxParser;
+  return bridgeTxParser.getBridgeTransactionsInThisBlock(web3Client, blockHashOrBlockNumber);
```
Comment (julia-zack) [inline lib/rsk-utils.js:411]: following some discussions we had about early return, we could refactor this to this:

```
  const isFingerroot500AlreadyActive = await Runners.common.forks.fingerroot500.isAlreadyActive();
  if (!isFingerroot500AlreadyActive) {
    return hopBridgeTxParser.getBridgeTransactionsInThisBlock(web3Client, blockHashOrBlockNumber);
  }

  const isLovell700AlreadyActive = await Runners.common.forks.lovell700.isAlreadyActive()
  if (!isLovell700AlreadyActive){
    return fingerrootBridgeTxParser.getBridgeTransactionsInThisBlock(web3Client, blockHashOrBlockNumber);
  }

  const bridgeTransactionParser = new BridgeTransactionParser(web3Client);
  return bridgeTransactionParser.getBridgeTransactionsInThisBlock(blockHashOrBlockNumber);
```

Comment (marcos-iov) [reply lib/rsk-utils.js:411]: I'd leave it as is, some day we'll get rid of the regression tests here and many of these checks won't be necessary anymore

File: lib/tests/2wp.js (hunk 2 above)
Comment (marcos-iov) [reply lib/tests/2wp.js:194]: Agree, too much stuff here not directly related to the test

**Comment by marcos-iov** [comment]
pipeline:run

**Review by marcos-iov — COMMENTED** [review]
Could we also check the existance of the event for a refund/migration?

**Review by jeremy-then — APPROVED** [review]
LGTM.

**Comment by marcos-iov** [comment]
pipeline:run

**Comment by nathanieliov** [comment]
pipeline:run

**Review by jeremy-then — APPROVED** [review]
LGTM.

File: lib/tests/2wp.js (hunk 2 above)
Comment (nathanieliov) [reply lib/tests/2wp.js]: Solved

**Review by marcos-iov — APPROVED** [review]

**Review by apancorb — APPROVED** [review]

**Pull request merged**